packages/design-system/src/Checkbox/Checkbox.styles.tsx

```diff
@@ -0,0 +1,165 @@
+import styled, { css } from "styled-components";
+
+const Variables = css`
+  --checkbox-color-label: var(--ads-v2-color-fg);
+  --checkbox-color-border: var(--ads-v2-color-border);
+  --checkbox-color-background-selected: var(--ads-v2-color-bg-brand-secondary);
+  --checkbox-color-border-selected: var(--ads-v2-color-bg-brand-secondary);
+  --checkbox-color-background-checkmark: var(
+    --ads-v2-color-fg-on-brand-secondary
+  );
+`;
+
+const Checkbox = css`
+  input[type="checkbox"] {
+    opacity: 0;
+    position: absolute;
+    left: 0;
+  }
+
+  & {
+    cursor: pointer;
+    position: relative;
+    padding-left: var(--ads-v2-spaces-7);
+    font-family: var(--ads-v2-font-family);
+    color: var(--checkbox-color-label);
+    display: flex;
+    align-items: center;
+    justify-content: center;
+  }
+
+  &::before {
+    content: "";
+    position: absolute;
+    left: 0;
+    top: var(--ads-v2-spaces-1);
+    width: var(--ads-v2-spaces-5);
+    height: var(--ads-v2-spaces-5);
+    border: 1px solid var(--ads-v2-color-border);
+    border-radius: var(--ads-v2-border-radius);
+    box-sizing: border-box;
+  }
+
+  &::after {
+    content: "";
+    position: absolute;
+    left: 0;
+    top: var(--ads-v2-spaces-1);
+    width: var(--ads-v2-spaces-5);
+    height: var(--ads-v2-spaces-5);
+    border-radius: var(--ads-v2-border-radius);
+    background-color: var(--checkbox-color-background-selected);
+    border: 1px solid var(--checkbox-color-border-selected);
+    box-sizing: border-box;
+    transform: scale(0);
+    transition: transform 0.2s ease;
+    z-index: 1;
+  }
+
+  label {
+    user-select: none;
+    cursor: pointer;
+  }
+
+  label::after {
+    content: "";
+    opacity: 0;
+    position: absolute;
+    left: 3px;
+    top: 11px;
+    width: 5px;
+    height: var(--ads-v2-spaces-1);
+    border-radius: 4px;
+    background-color: var(--checkbox-color-background-checkmark);
+    transform: rotateZ(48deg);
+    z-index: 2;
+  }
+
+  input[type="checkbox"]:checked + label::after {
+    opacity: 1;
+  }
+
+  label::before {
+    content: "";
+    opacity: 0;
+    position: absolute;
+    left: 4px;
+    top: 9px;
+    width: 10px;
+    height: var(--ads-v2-spaces-1);
+    border-radius: 4px;
+    background-color: var(--checkbox-color-background-checkmark);
+    transform: rotateZ(304deg);
+    z-index: 2;
+  }
+
+  input[type="checkbox"]:checked + label::before {
+    opacity: 1;
+  }
+`;
+
+export const StyledCheckbox = styled.label<{
+  isChecked: boolean;
+  isIndeterminate?: boolean;
+  isFocusVisible?: boolean;
+  isDisabled?: boolean;
```

Comment:
I would really like to do isDisabled to follow  the pattern of using the `is` prefix for all booleans, but we've had this discussion and @rohitagarwal88 recommended using the html standard. Given that in certain cases react-aria forces us to use `isDisabled`, and in the interest of keeping a consistent API for our components, can we please adopt this pattern as well? 

This will entail making changes to https://github.com/appsmithorg/design-system/pull/311 to comply.